tests/conftest.py

```diff
@@ -238,3 +239,38 @@ def unprivileged_client(
 
         else:
             yield admin_client
+
+
+@pytest.fixture(scope="session")
+def dsc_resource(admin_client: DynamicClient, name: str = "default-dsc"):
+    for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
+        yield dsc
+
+
+def get_dsc_component_state(dsc_resource: DataScienceCluster, component_name: str) -> str:
+    return dsc_resource.instance.spec.components[component_name].managementState
+
+
+@pytest.fixture(scope="module")
+def set_dsc_component_state(
+    dsc_resource: DataScienceCluster,
+    request: FixtureRequest,
+) -> Generator[DataScienceCluster, Any, Any]:
+    component_name = getattr(request.module, "component_name")
+    desired_state = getattr(request.module, "desired_state")
```

Comment:
when more than one param is needed, please use pytest.params with a dict to pass values so it is clear when calling the fixture what is passed
e.g

@pytest.mark.parametrize(
    "fixture_name",
[
        pytest.param(
            {"component_name": "xxxx", "desired_state": "yyy"},
        )
    ],
    indirect=True,
)